File: src/main/java/io/github/gravetii/scene/settings/GameTimeScene.java
package io.github.gravetii.scene.settings;

import io.github.gravetii.scene.FxDimensions;
import io.github.gravetii.scene.FxScene;

import javafx.geometry.Dimension2D;
import javafx.stage.Stage;

import java.util.Optional;

public class GameTimeScene extends FxScene {
  private final GameTimeComponent component;

  public GameTimeScene(Stage stage) throws Exception {
    super(stage);
    this.component = new GameTimeComponent(stage);
  }

  @Override
  protected void build() {
    this.showCenter(this.component);
  }

  @Override
  protected String title() {
    return "Configure game time";
  }

  @Override
  protected Optional<FxDimensions> preferredDimensions() {
    FxDimensions dimensions = new FxDimensions(new Dimension2D(500, 180));
    return Optional.of(dimensions);
  }
}

File: src/main/kotlin/io/github/gravetii/scene/settings/GameTimeScene.kt
package io.github.gravetii.scene.settings

import io.github.gravetii.scene.FxDimensions
import io.github.gravetii.scene.FxScene
import javafx.geometry.Dimension2D
import javafx.stage.Stage

class GameTimeScene(stage: Stage) : FxScene(stage) {

    private val component = GameTimeComponent(stage)

    override fun build() = showCenter(component)

    override fun title() = "Configure game time"

    override fun preferredDimensions(): FxDimensions = FxDimensions(Dimension2D(500.0, 180.0))

}